### PR TITLE
[SPARK-37794][BUILD] Remove internal log4j bridge api usage

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

This patch removes log4j 1.x bridge api usage.

### Why are the changes needed?

Although for compatibility reason, we include log4j 1.x bridge api dependency. But for Spark itself, it is better to remove the bridge api usage.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Pass all tests

### CODE REVIEW

**Comment by viirya** [comment]
Two log4j 1.x bridge API usage I've not removed yet:

`TestUtils.configTestLog4j` and `python/pyspark/testing/utils.py`.

File: core/src/main/scala/org/apache/spark/internal/Logging.scala [hunk 1]
```diff
@@ -237,6 +231,19 @@ private[spark] object Logging {
     "org.apache.logging.slf4j.Log4jLoggerFactory".equals(binderClass)
   }
 
+  /**
+   * Return true if log4j2 is initialized by default configuration which has one
+   * appender with error level. See `org.apache.logging.log4j.core.config.DefaultConfiguration`.
+   */
+  private[spark] def islog4j2DefaultConfigured(): Boolean = {
+    val rootLogger = LogManager.getRootLogger.asInstanceOf[org.apache.logging.log4j.core.Logger]
```
Comment (dongjoon-hyun) [inline core/src/main/scala/org/apache/spark/internal/Logging.scala:239]: Can we use `import org.apache.logging.log4j.core.Logger` explicitly?

File: sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/HiveMetastoreLazyInitializationSuite.scala [hunk 2]
```diff
@@ -30,7 +30,8 @@ class HiveMetastoreLazyInitializationSuite extends SparkFunSuite {
       .enableHiveSupport()
       .config("spark.hadoop.hive.metastore.uris", "thrift://127.0.0.1:11111")
       .getOrCreate()
-    val originalLevel = org.apache.log4j.Logger.getRootLogger().getLevel
+    val originalLevel = org.apache.logging.log4j.LogManager.getRootLogger
```
Comment (dongjoon-hyun) [inline sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/HiveMetastoreLazyInitializationSuite.scala:33]: Can we use `import org.apache.logging.log4j.LogManager` explicitly?

File: sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/HiveMetastoreLazyInitializationSuite.scala [hunk 3]
```diff
@@ -30,7 +30,8 @@ class HiveMetastoreLazyInitializationSuite extends SparkFunSuite {
...
+      .asInstanceOf[org.apache.logging.log4j.core.Logger].getLevel
```
Comment (dongjoon-hyun) [inline sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/HiveMetastoreLazyInitializationSuite.scala:34]: ditto.

**Comment by dongjoon-hyun** [comment]
cc @williamhyun

**Review by dongjoon-hyun — DISMISSED** [review]
+1, LGTM (Pending CI). Thank you for updates.

**Comment by dongjoon-hyun** [comment]
Hive module seems to fail. Could you re-trigger GitHub Action?

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHive.scala [hunk 4]
```diff
@@ -535,10 +535,12 @@ private[hive] class TestHiveSparkSession(
   def reset(): Unit = {
     try {
       // HACK: Hive is too noisy by default.
-      org.apache.log4j.LogManager.getCurrentLoggers.asScala.foreach { log =>
-        val logger = log.asInstanceOf[org.apache.log4j.Logger]
+      org.apache.logging.log4j.LogManager.getContext(false)
```
Comment (dongjoon-hyun) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHive.scala:538]: Hive tests seem to fail consistently on this PR. Do you think this is relevant, @viirya ?

BTW, is the following PR relevant to this, too?
- https://github.com/apache/spark/pull/35080

**Review by dongjoon-hyun — COMMENTED** [review]
Hive tests

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHive.scala (hunk 4 above)
Comment (viirya) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHive.scala:538]: #35080 is different.

Comment (dongjoon-hyun) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHive.scala:538]: Got it.

Then, @viirya , does Hive test pass in your local environment?

Comment (viirya) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHive.scala:538]: Let me check. This changes logger levels. It shouldn't affect tests unrelated to logging).

Comment (viirya) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHive.scala:538]: e.g.,
```
 [info] - Query with constant folding the CAST *** FAILED *** (6 milliseconds)
[info]   org.apache.spark.sql.AnalysisException: Table or view not found: src; line 1 pos 50;
```

Comment (viirya) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHive.scala:538]: Oops, I made a mistake...let me update...

Comment (viirya) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHive.scala:538]: It should pass now.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (Pending CIs). Thank you for updating.

**Comment by viirya** [comment]
Thank you @dongjoon-hyun

**Comment by dongjoon-hyun** [comment]
Merged to master.

**Pull request closed without merging**